uwblueprint/extend-a-family
# Match Component Prototype

## Notion ticket link

[Matching Element Authoring](https://www.notion.so/uwblueprintexecs/Prototype-Matching-Element-Authoring-eec27da66f0640599b6c412c100db451?pvs=4)

## What's Next?
* Developing a 'Picture' Component, but I think this should be a separate ticket / PR.
* Add a 'Clear all edges button', if wanted.
* Give user choice if they want the node to have a border or not.
* Try to find a solution to the bug where you select two nodes but nothing happens, from my testing it seems to be a race condition issue? If you go slower, you don't run into it...
* There are a LOT of linter issues .. :(


## Implementation description
* Developed matching component and corresponding edit panel



## Steps to test
1. Navigate to 'Create Module' Page.
1. Drag a 'Match' component into your grid.
2. Add X rows and Y columns, this will populate your match component with blank nodes.
3. Click on a node, then click on another. This should draw a blue line between them
a. Nodes cannot be connected vertically, only horizontally.
b. Sometimes you will run into a bug where nothing happens, try clicking the second node of the sequence again.
4. Touch two nodes that have an existing connection in a sequence. This should remove the edge.
5. Click on any node, then go to the edit panel and play around with its width or height, this should adjust the size of the particular node.
6. Click on any node, make it a text box and try to edit it.

#### It may help to switch from edit mode to view mode when playing with the component.


## What should reviewers focus on?
* How does playing around with the component feel? If you did not have instructions would it be intuitive? What changes would you make?


## Checklist
- [X ] My PR name is descriptive and in imperative tense
